javaparser-core/src/main/java/com/github/javaparser/utils/SourceRoot.java

```diff
@@ -50,6 +46,9 @@
     private JavaParser javaParser = new JavaParser();
 
     public SourceRoot(Path root) {
+        if (Files.isRegularFile(root)) {
```

Comment:
I think your right! What I wanted to say is: it must be a directory, so writing `!directory` is more cleaner to read, thx